tap_hubspot/__init__.py

```diff
@@ -864,9 +861,13 @@ def do_sync(STATE, catalogs):
 
 class Context(object):
     def __init__(self, catalog):
-        self.selected_stream_ids = set(
-            s.get('tap_stream_id') for s in catalog.get('streams')
-            if s.get('schema').get('selected'))
+        self.selected_stream_ids = set()
+
+        for stream in catalog.get('streams'):
+            for mdata in stream['metadata']:
+                if mdata['breadcrumb'] == [] and mdata['metadata']['selected'] == True:
+                    self.selected_stream_ids.add(stream.get('tap_stream_id'))
```

Comment:
This should always be present, and it might be better to blow up rather than to have a `tap_stream_id` of None. Should be `stream['tap_stream_id']`